packages/svelte/src/internal/client/runtime.js

```diff
@@ -158,53 +154,18 @@ export function is_dirty(reaction) {
 		return true;
 	}
 
+	if (flags & DERIVED) {
+		reaction.f &= ~WAS_MARKED;
```

Comment:
microscopically more efficient:

```suggestion
	if ((flags & WAS_MARKED) !== 0) {
		reaction.f ^= WAS_MARKED;
```

Reply:
errr... except I made this change locally and tests failed. which is strange

Reply:
(separately: I get a bit nervous about doing this here. `is_dirty` doesn't necessarily touch every derived, it will return as soon as it knows that a reaction is dirty)

Reply:
> So you will traverse upwards and unmark them all,

That's the bit I'm not sure about — in `is_dirty` we have this code...

```js
for (var i = 0; i < length; i++) {
	var dependency = dependencies[i];

	if (is_dirty(/** @type {Derived} */ (dependency))) {
		update_derived(/** @type {Derived} */ (dependency));
	}

	if (dependency.wv > reaction.wv) {
		return true;
	}
}
```

...which basically means that if the first dependency is known to be dirty, we don't bother checking the second one. Is there a scenario in which that second dependency doesn't get read during the subsequent effect run, and is thus incorrectly left marked? I haven't managed to produce a failing test, but it feels like a possibility.

> did you bench it?

No but

1. the `!==` means that the code will be compiled more efficiently (because the engine knows that it is dealing with a boolean)
2. it's one bitwise operation rather than two